sympy/series/limits.py

```diff
@@ -212,36 +211,50 @@ def doit(self, **hints):
         if not e.has(z):
             return e
 
+        try:
+            if e.is_meromorphic(z, z0):
```

Comment:
Did `e.is_meromorhpic(z, z0)` raise? If not, then this could come before `try` and the `try-except` block would not be always created.

Reply:
Ok.

Reply:
There is far too much code in between the try and the except here. Use of `try/except` should be limited to precisely the line/call that is expected to raise. Probably there is a better way of handling the exceptional cases here without using try/except at all.

Reply:
`coeff, exp = newe.leadterm(z)` This is the line expected to raise

Reply:
What would `newe` need to be in order for this to raise?

Reply:
I don't think it will be easy to generalise, though we might succeed in doing so, but since there can be always new cases, I might avoid this.

Reply:
Can you give any example?

Reply:
Terms like `atan(1/x)`.

Reply:
Isn't that a bug?
```python
>>> from sympy import *
>>> x = Symbol('x')
>>> 
>>> atan(1/x).series(x)
pi/2 - x + x**3/3 - x**5/5 + O(x**6)
>>> atan(1/x).series(x).leadterm(x)
(pi/2, 0)
>>> atan(1/x).leadterm(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/enojb/current/sympy/sympy/sympy/core/expr.py", line 3412, in leadterm
    should have been free of %s but got %s""" % (self, x, x, c)))
ValueError: 
cannot compute leadterm(atan(1/x), x). The coefficient should have
been free of x but got atan(1/x)
```

Reply:
Seems so.

Reply:
Until these issues get fixed, will have to keep the `try-except` block.